maths/series/arithmetico_geometric_sequence.py

```diff
@@ -12,134 +12,198 @@
+                series.append(f"({ap_part}) x {gp_part}")
         return series
 
-    def last_term_value(self):
-        return (self.a + (self.n - 1) * self.d) * (self.b * pow(self.r, self.n - 1))
-
-    def sum(self):
+    def last_term_value(self) -> float:
+        """
+        >>> AGS(2, 3, 5, 6, 3).last_term_value()
+        1440
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).last_term_value()
+        1.4976000000000003
+        >>> AGS(9, 3, 2, -0.5, 0).last_term_value()
+        'None'
+        """
+        if self.term_count == 0:
+            return "None"
+        return (
+            self.arithmetic_value + (self.term_count - 1) * self.arithmetic_difference
+        ) * (self.geometric_value * pow(self.common_ratio, self.term_count - 1))
+
+    def sum(self) -> float:
+        """
+        >>> AGS(2, 3, 5, 6, 3).sum()
+        1600.0
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).sum()
+        8.049600000000002
+        >>> AGS(9, 3, 2, -0.5, 0).sum()
+        'None'
+        """
+        if self.term_count == 0:
+            return "None"
         return (
-            (self.a * self.b)
-            - ((self.a + (self.n * self.d)) * (self.b * pow(self.r, self.n)))
-        ) / (1 - self.r) + (self.d * self.b * self.r * (1 - pow(self.r, self.n))) / pow(
-            (1 - self.r), 2
+            (self.arithmetic_value * self.geometric_value)
+            - (
+                (self.arithmetic_value + (self.term_count * self.arithmetic_difference))
+                * (self.geometric_value * pow(self.common_ratio, self.term_count))
+            )
+        ) / (1 - self.common_ratio) + (
+            self.arithmetic_difference
+            * self.geometric_value
+            * self.common_ratio
+            * (1 - pow(self.common_ratio, self.term_count))
+        ) / pow(
+            (1 - self.common_ratio), 2
         )
 
-    def inf_sum(self):
-        return (self.a * self.b) / (1 - self.r) + (self.d * self.b * self.r) / pow(
-            (1 - self.r), 2
+    def inf_sum(self) -> float:
+        """
+        >>> AGS(2, 3, 5, 6, 3).inf_sum()
+        1.6
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).inf_sum()
+        9.533333333333335
+        >>> AGS(9, 3, 2, -0.5, 0).inf_sum()
+        10.666666666666666
+        """
+        return (self.arithmetic_value * self.geometric_value) / (
+            1 - self.common_ratio
+        ) + (
+            self.arithmetic_difference * self.geometric_value * self.common_ratio
+        ) / pow(
+            (1 - self.common_ratio), 2
         )
 
-    def nth_term_value(self, k):
-        return (self.a + (k - 1) * self.d) * (self.b * pow(self.r, k - 1))
+    def kth_term_value(self, k_in_kth_term: int) -> float:
+        """
+        >>> AGS(2, 3, 5, 6, 3).kth_term_value(2)
+        150
+        >>> AGS(3, 2.1, 1.3, 0.4, 3).kth_term_value(1)
+        3.9000000000000004
+        >>> AGS(9, 3, 2, -0.5, 0).kth_term_value(0)
+        'None'
+        """
+        if k_in_kth_term == 0:
+            return "None"
```

Comment:
The type hint promises that this function returns a float, not a str.  If we cannot calculate a float to return then we could raise a ValueError.